lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1660,6 +1660,78 @@ struct HandshakeAnalysisPass
 };
 } // namespace
 
+// This pass iteratively checks the use count of each value, and reformats the
+// MLIR such that every value has exact one use. If a value has multiple uses, a
+// fork is inserted. If a value has no use, a sink is inserted.
+void insertOpForExactOneUse(Value value, handshake::FuncOp funcOp,
+                            OpBuilder &rewriter) {
+
+  if (value.use_empty())
+    rewriter.create<SinkOp>(funcOp.getLoc(), value);
+  else {
+    auto uses = value.getUses();
+    auto outs = std::distance(cbegin(uses), cend(uses));
+    std::vector<Operation *> ops;
+    std::vector<unsigned> idx;
+    for (auto &use : uses) {
+      ops.push_back(use.getOwner());
+      idx.push_back(use.getOperandNumber());
+    }
+
+    // There is an unkown issue when forking the function arguments directly
+    // The temporary solution to to create merge followed by the fork, and
+    // remove the merge after creating fork
+    rewriter.setInsertionPoint(
+        &(funcOp.getOperation()->getRegions().begin()->begin()->back()));
```

Comment:
I'd recommend adding a helper method to HandShake::FuncOp for `funcOp.getOperation()->getRegions().begin()->begin()`